src/events/neo4j.js

```diff
@@ -0,0 +1,387 @@
+    const { host, port } = getAddressInfo(session);
+
+    const {
+        metadata, fullMetadata,
+    } = getNeo4JSessionMetadata(session);
+
+    metadata.port = port;
+
+    const resource = new serverlessEvent.Resource([
+        host,
+        'neo4j',
+        metadata.operation,
+    ]);
+
+    const dbApiEvent = new serverlessEvent.Event([
+        `neo4j-${uuid4()}`,
+        utils.createTimestampFromTime(startTime),
+        null,
+        'neo4j',
+        0,
+        errorCode.ErrorCode.OK,
+    ]);
+
+    dbApiEvent.setResource(resource);
+
+    eventInterface.addToMetadata(dbApiEvent, metadata, fullMetadata);
+
+    return dbApiEvent;
+}
+
+
+/**
+ * Wraps neo4j.Session.run functions with tracing
+ * @param {Function} wrappedFunction The function to wrap from Neo4j
+ * @returns {Function} The wrapped function
+ */
+function neo4jSessionRunWrapper(wrappedFunction) {
+    return function internalNeo4jSessionRunWrapper(...args) {
+        const relevantArgs = getArgsFromFunction(...args, this.constructor.name);
+        const {
+            query, params, transactionConfig,
+        } = relevantArgs;
+
+        utils.debugLog('User called Neo4j wrapped Session run function');
+
+        const resultResponse = wrappedFunction.apply(
+            this,
+            [query, params, transactionConfig]
+        );
+
+        try {
+            const startTime = Date.now();
+
+            const dbApiEvent = createNewNeo4jEvent(this, startTime);
+
+            eventInterface.addToMetadata(
+                dbApiEvent,
+                {},
+                { query, param: params, transaction_config: transactionConfig }
+            );
+
+            const originalSubscribe = resultResponse.subscribe;
+
+            // Override the Result subscribe with patched subscriber
+            // eslint-disable-next-line func-names
+            resultResponse.subscribe = function (observer) {
+                originalSubscribe.call(this, getPatchedSubscribeMethods(
+                    this,
+                    observer,
+                    arguments,
+                    dbApiEvent,
+                    startTime
+                ));
+            };
+
+            return resultResponse;
+        } catch (error) {
+            tracer.addException(error);
+            return resultResponse;
+        }
+    };
+}
+
+
+/**
+ * Wraps neo4j.Transaction.run functions with tracing
+ * @param {Function} wrappedFunction The function to wrap from Neo4j
+ * @returns {Function} The wrapped function
+ */
+function neo4jTransactionRunWrapper(wrappedFunction) {
+    return function internalNeo4jTransactionRunWrapper(...args) {
+        const relevantArgs = getArgsFromFunction(...args, this.constructor.name);
+        const {
+            query, params,
+        } = relevantArgs;
+
+        utils.debugLog('User called Neo4j wrapped Transaction run function');
+
+        const resultResponse = wrappedFunction.apply(this, [query, params]);
```

Comment:
Now if the customer's code throws an exception we will not add it to the tracer object

Reply:
fixed